### PR TITLE
:warning: Separate Target and RuleSet models/apis

### PR DESCRIPTION
**Pull request by mansam**

### CODE REVIEW

File: api/target.go [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+
+//
+// TargetHandler handles Target resource routes.
+type TargetHandler struct {
+	BaseHandler
+}
+
+func (h TargetHandler) AddRoutes(e *gin.Engine) {
+	routeGroup := e.Group("/")
+	routeGroup.Use(Required("rulesets"), Transaction)
```
Comment (jortel) [inline api/target.go:26]: targets
Also, needs to be added to the realm.

File: api/target.go [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+
+//
+// TargetHandler handles Target resource routes.
+type TargetHandler struct {
+	BaseHandler
+}
+
+func (h TargetHandler) AddRoutes(e *gin.Engine) {
+	routeGroup := e.Group("/")
+	routeGroup.Use(Required("targets"), Transaction)
```
Comment (jortel) [inline api/target.go:26]: This is probably practical but expected the Transaction only on:
Create
Update
Delete

File: migration/v8/model/target.go [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+//
+// Target - analysis rule selector.
+type Target struct {
+	Model
+	UUID        *string `gorm:"uniqueIndex"`
+	Name        string  `gorm:"uniqueIndex;not null"`
+	Description string
+	Choice      bool
+	Labels      JSON
```
Comment (jortel) [inline migration/v8/model/target.go:11]: Missing: `gorm:"type:json"`

File: api/target.go (hunk 2 above)
Comment (mansam) [reply api/target.go:26]: I think we've been using it like this everywhere, and relying on the method check in the Transaction handler unless I'm mistaken.

File: api/ruleset.go [hunk 4]
```diff
@@ -268,13 +309,10 @@ func (r *RuleSet) With(m *model.RuleSet) {
 // Model builds a model.
 func (r *RuleSet) Model() (m *model.RuleSet) {
 	m = &model.RuleSet{
-		Kind:        r.Kind,
-		Name:        r.Name,
-		Description: r.Description,
```
[the remark this answers is not in the document]
Comment (mansam) [reply api/ruleset.go:273]: Yeah, good idea.

**Review by jortel — APPROVED** [review]
Looks great!

File: api/ruleset.go [hunk 5]
```diff
@@ -71,7 +75,16 @@ func (h RuleSetHandler) List(ctx *gin.Context) {
 		h.DB(ctx),
 		clause.Associations,
 		"Rules.File")
-	result := db.Find(&list)
+
+	filter, err := qf.New(ctx,
+		[]qf.Assert{
+			{Field: "labels", Kind: qf.STRING},
```
Comment (jortel) [inline api/ruleset.go:81]: Adding `name` to the filter in the interest of completeness _may_ be a nice to have but can always add later.

File: api/target.go (hunk 2 above)
Comment (jortel) [reply api/target.go:26]: You're correct.  I forgot.

**Pull request merged**